### PR TITLE
Feat: allow delete the provisioning resources

### PR DESCRIPTION
**Pull request by chivalryq**
Signed-off-by: Qiaozp <qiaozhongpei.qzp@alibaba-inc.com>

Add a feature gate: `AllowDeleteProvisioningResource`. When enabled, controller will allow to delete Configuration in the state of `ProvisioningAndChecking`. This is useful when developing new configuration. 

Here's some examples. 

- When apply a large Configuration which creates more than one cloud resource, it could fail to create all resources but some of resources have been created. 
- When we apply a Configuration and it fails somehow. (Like the API has been changed by vendor)

Before, we have no method to delete this Configuration other than delete the `finalizer` fields manually and delete the created resources on vendor's console, which is tedious and annoying. 

Now created resource can be deleted by controller automatically.

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -25,3 +25,9 @@ backend:
   namespace: vela-system
 
 githubBlocked: "'false'"
+
+featureGates:
+  # Enable the feature of allowing to delete a configuration whose cloud resources is not fully provisioned, or error happens
+  # This guarantees that the partial cloud resources will be deleted when the configuration is deleted
+  # Default value is true
+  allowDeleteHalfway: true
```
Comment (wonderflow) [inline chart/values.yaml:33]:
```suggestion
  allowDeleteProvisioningResource: true
```

Comment (chivalryq) [reply chart/values.yaml]: Done

File: chart/values.yaml [hunk 2]
```diff
@@ -15,13 +15,19 @@ jobNodeSelector: ""
 
 resources:
   limits:
-    cpu: "1000m"
-    memory: "2Gi"
+    cpu: "500m"
+    memory: "1Gi"
```
Comment (wonderflow) [inline chart/values.yaml:19]: you can keep the limits quota, just reduce the requests.

File: chart/values.yaml [hunk 3]
```diff
@@ -15,13 +15,19 @@ jobNodeSelector: ""
...
   requests:
-    cpu: "1000m"
-    memory: "2Gi"
+    cpu: "500m"
+    memory: "1Gi"
```
Comment (wonderflow) [inline chart/values.yaml:22]: can we make it even smaller?

**Review by wonderflow — APPROVED** [review]
generally LGTM

File: chart/values.yaml (hunk 3 above)
Comment (chivalryq) [reply chart/values.yaml:22]: Yes, this is for pod where run `terraform apply` which doesn't require much resources

**Pull request merged**